curriculum/challenges/english/07-scientific-computing-with-python/learn-tree-traversal-by-building-a-binary-search-tree/65c644829cfb63acf3479d09.md

```diff
@@ -11,16 +11,20 @@ Insert the nodes in the list `nodes` into the `bst` instance by iterating over t
 
 # --hints--
 
-You should have a `for node in nodes` loop.
+You should use a properly structured `for` loop with `node` as the iteration variable and call `bst.insert(node)` inside the loop.
 
 ```js
-assert.match(code, /for\s+node\s+in\s+nodes/);
-```
-
-You should have `bst.insert(node)` inside the `for` loop.
-
-```js
-assert.match(code, /bst\.insert/);
+({
+  test: () =>
+    assert(
+      runPython(`
+        _Node(_code)
+        .find_for("node", "nodes")  
+        .find_body()  
+        .has_call("bst.insert(node)")
```

Comment:
you may want to test. You need to put it all in one line or wrap it in `()`, same for the other one